libcudacxx/include/cuda/std/__cccl/compiler.h

```diff
@@ -11,25 +11,31 @@
 #ifndef __CCCL_COMPILER_H
 #define __CCCL_COMPILER_H
 
-// Determine the host compiler
+// Determine the host compiler and its version
 #if defined(__INTEL_LLVM_COMPILER)
 #  define _CCCL_COMPILER_ICC_LLVM
+#  define _CCCL_COMPILER_ICC_LLVM_VERSION __INTEL_LLVM_COMPILER
 #elif defined(__INTEL_COMPILER)
 #  define _CCCL_COMPILER_ICC
 #elif defined(__NVCOMPILER)
 #  define _CCCL_COMPILER_NVHPC
+#  define _CCCL_COMPILER_NVHPC_VERSION \
+    (__NVCOMPILER_MAJOR__ * 10000 + __NVCOMPILER_MINOR__ * 100 + __NVCOMPILER_PATCHLEVEL__)
+static_assert(__NVCOMPILER_PATCHLEVEL__ < 100, "__NVCOMPILER_PATCHLEVEL__ must be less than 100");
 #elif defined(__clang__)
 #  define _CCCL_COMPILER_CLANG
 #  define _CCCL_CLANG_VERSION (__clang_major__ * 10000 + __clang_minor__ * 100 + __clang_patchlevel__)
+static_assert(__clang_patchlevel__ < 100, "__clang_patchlevel__ must be less than 100");
 #elif defined(__GNUC__)
 #  define _CCCL_COMPILER_GCC
 #  define _CCCL_GCC_VERSION (__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__)
+static_assert(__GNUC__ < 100);
+static_assert(__GNUC_MINOR__ < 100);
+static_assert(__GNUC_PATCHLEVEL__ < 100);
 #elif defined(_MSC_VER)
 #  define _CCCL_COMPILER_MSVC
 #  define _CCCL_MSVC_VERSION      _MSC_VER
 #  define _CCCL_MSVC_VERSION_FULL _MSC_FULL_VER
-#elif defined(__IBMCPP__)
-#  define _CCCL_COMPILER_IBM
```

Comment:
Sorry for being lazy and doing this as a second step: I just grepped for `_CCCL_COMPILER_IBM` and there are a few occurences. I think we should remove all traces of `_CCCL_COMPILER_IBM` throughout the codebase if we remove this macro here.

But maybe, let's have somebody else on the team confirm that IBMCPP support is really dead and this is not causing any kind of breakage. Maybe it's also better to pull the plug on IBMCPP in a separate PR.

Reply:
there is a tricky point here. PowerPC architecture is removed in CUDA 12.5 https://docs.nvidia.com/cuda/archive/12.5.0/cuda-toolkit-release-notes/index.html#deprecated-or-dropped-architectures but CCCL is compatible with CUDA 11.x and 12.x. Probably, we cannot remove it yet

Reply:
Result of an internal discussion: we can safely remove support for IBMCPP. You can choose whether you want to do that in this PR or a separate PR. But please remove all traces of `_CCCL_COMPILER_IBM`.